model-optimizer/extensions/front/mxnet/deduce_parameter_shape.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+from typing import List
+
+import networkx as nx
+import numpy as np
+import logging as log
+from mo.front.common.partial_infer.utils import dynamic_dimension, shape_array, unmask_shape
+from mo.front.common.replacement import FrontReplacementSubgraph
+from mo.graph.graph import Graph
+from mo.graph.graph import Node
+
+
+class DeduceParameterShape(FrontReplacementSubgraph):
+    enabled = True
+    graph_condition = [lambda graph: graph.graph['fw'] == 'mxnet' and graph.graph['user_shapes'] is None]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        convolutions = graph.get_op_nodes(op='Convolution')
+        parameter_nodes = graph.get_op_nodes(op='Parameter')
+        for parameter in parameter_nodes:
+            conv = find_nearest_conv(graph, parameter, convolutions)
+            if conv.in_port(1).disconnected() or not conv.in_node(1).has_valid('value'):
+                continue
+
+            conv_shape = conv.in_node(1).value.shape
+            parameter_rank = len(conv_shape)
+            num_space_dimensions = parameter_rank - 2
+            num_channels = conv_shape[1]
+            parameter_shape = [dynamic_dimension, num_channels, *([dynamic_dimension] * num_space_dimensions)]
+            parameter['shape'] = shape_array(parameter_shape)
```

Comment:
This solution is wrong for a number of reasons:

1. The value of the convolution weightsmay not be constant. For example, there are models where they are inferred at runtime.
2. The idea to get the second dimension of the weights is wrong as well. At this stage of the model conversion the Convolution operation does not correspond to its opset specification. Take a look at the shape inference of the Convolution operation and you will see a lot of manipulations with the operation attributes, including the weights tensor manupulation.